Tests/PHPUnit/Subscriber/PaymentMethodSubscriberTest.php

```diff
@@ -0,0 +1,126 @@
+use MollieShopware\Components\Helpers\MollieShopSwitcher;
+use MollieShopware\MollieShopware;
+use MollieShopware\Subscriber\PaymentMethodSubscriber;
+use MollieShopware\Tests\PHPUnit\Utils\Fakes\View\FakeView;
+use PHPUnit\Framework\TestCase;
+use Psr\Log\LoggerInterface;
+
+class PaymentMethodSubscriberTest extends TestCase
+{
+    const SHOPWARE_PAYMENT_METHOD_NAME = 'debit';
+
+    /**
+     * Creates an array of Mollie payment method objects.
+     *
+     * @param array $methods
+     * @return array
+     */
+    public function setUpAvailableMethods(array $methods)
+    {
+        $client = $this->createMock(MollieApiClient::class);
+
+        $methodObjects = [];
+
+        foreach ($methods as $method) {
+            $methodObject = new Method($client);
+            $methodObject->id = $method;
+
+            $methodObjects[] = $methodObject;
+        }
+
+        return $methodObjects;
+    }
+
+    /**
+     * Creates a view with assigned payment methods.
+     *
+     * @return FakeView
+     */
+    public function setUpView()
+    {
+        $view = new FakeView($this->createMock(Enlight_Template_Manager::class));
+
+        $view->assign('sPayments', [
+            [
+                'name' => self::SHOPWARE_PAYMENT_METHOD_NAME,
+            ],
+            [
+                'name' => MollieShopware::PAYMENT_PREFIX . PaymentMethod::CREDITCARD,
+            ],
+            [
+                'name' => MollieShopware::PAYMENT_PREFIX . PaymentMethod::IDEAL,
+            ],
+            [
+                'name' => MollieShopware::PAYMENT_PREFIX . PaymentMethod::KLARNA_PAY_LATER,
+            ],
+            [
+                'name' => MollieShopware::PAYMENT_PREFIX . PaymentMethod::PAYPAL,
+            ]
+        ]);
+
+        return $view;
+    }
+
+    /**
+     * This test verifies that our required subscribers
+     * are not changed without recognizing it.
+     *
+     * @testdox Method getSubscribedEvents returns the expected events.
+     */
+    public function testSubscribedEvents()
+    {
+        $expected = [
+            'Enlight_Controller_Action_PostDispatchSecure_Frontend_Checkout',
+        ];
+
+        $this->assertEquals($expected, array_keys(PaymentMethodSubscriber::getSubscribedEvents()));
+    }
+
+    /**
+     * This test verifies if payment methods that are unavailable
+     * are removed from the checkout view.
+     *
+     * @testdox Method removeUnavailablePaymentMethods removes the expected payment methods from view.
+     */
+    public function testRemovesPaymentMethods()
+    {
+        $available = $this->setUpAvailableMethods(['ideal', 'paypal']);
+
+        $paymentMethodSubscriber = new PaymentMethodSubscriber(
+            $this->createMock(LoggerInterface::class),
+            $this->createMock(MollieShopSwitcher::class)
+        );
+
+        $view = $this->setUpView();
+
+        $paymentMethodSubscriber->removeUnavailablePaymentMethods(
```

Comment:
attention, this is not really testing what this test case implies.
i would think that this test verifies that our subscriber correctly removes the payment methods from the view.
but this actually just tests the "sub component" to remove methods according to a list of available payment methods.
if i would just remove the calling line in the subscriber itself, the test is still green, while the storefront/feature is broken.

so either you re-model it to test only the removing functionality, or you calll the "onFrontendCheckoutPostDispatch" function which might be the better one.
then you can also change the function to private inside the subscriber